algebird-core/src/main/scala/com/twitter/algebird/MapAlgebra.scala

```diff
@@ -222,4 +224,31 @@ object MapAlgebra {
 
   def dot[K, V](left: Map[K, V], right: Map[K, V])(implicit mring: Ring[Map[K, V]], mon: Monoid[V]): V =
     Monoid.sum(mring.times(left, right).values)
+
+  def cube[K, V](it: TraversableOnce[(K, V)])(implicit c: Cuber[K]): Map[c.K, TraversableOnce[V]] = {
+    val sg = new Semigroup[Iterable[V]] {
+      def plus(x: Iterable[V], y: Iterable[V]) = x ++ y
+    }
+    cubeSum(it.map { case (k, v) => (k, Iterable(v)) })(c, sg)
```

Comment:
I don't think this approach is going to work well (the default Iterable is List, I think, so you're going to get repeated ++ on List which gives an O(N^2) algorithm).

You're better off doing something like:
```scala
val map: collection.mutable.Map[c.K, List[V]] = collection.mutable.Map[c.K, List[V]]()
it.toIterator.foreach { case (k, v) =>
  c(k).foreach { ik =>
    map.get(ik) match {
     case Some(vs) => map += ik -> (v :: vs)
     case None => map += ik -> List(v)
    }
  }
  // now reverse the lists and return an immutable map.
  // you might just wrap this one with the immutable map that wraps a mutable one (so we
  // don't pay the immutable construction cost unless we need to, i.e. the user calls + or .toMap).
}
```